sqlmesh/core/console.py

```diff
@@ -927,6 +932,52 @@ def _confirm(self, message: str, **kwargs: t.Any) -> bool:
         self._print(message)
         return super()._confirm("", **kwargs)
 
+    def start_snapshot_progress(
+        self, snapshot: Snapshot, total_batches: int, environment: str
+    ) -> None:
+        """Indicates that a new load progress has begun."""
+        if not self.evaluation_batch_progress.get(snapshot.name):
+            view_name = snapshot.qualified_view_name.for_environment(environment)
+            self.evaluation_batch_progress[snapshot.name] = (view_name, 0, total_batches)
+            print(f"Starting '{view_name}', Total batches: {total_batches}")
+
+    def update_snapshot_progress(self, snapshot_name: str, num_batches: int) -> None:
+        """Update snapshot progress."""
+        view_name, loaded_batches, total_batches = self.evaluation_batch_progress[snapshot_name]
+        loaded_batches += 1
+        self.evaluation_batch_progress[snapshot_name] = (view_name, loaded_batches, total_batches)
+        finished_loading = loaded_batches == total_batches
+        status = "Loaded" if finished_loading else "Loading"
+        print(f"{status} '{view_name}', Completed Batches: {loaded_batches}/{total_batches}")
+        if finished_loading:
+            total_finished_loading = len(
+                [x for x in self.evaluation_batch_progress.values() if x[1] == x[2]]
```

Comment:
don't need the brackets, leave it a gen

Reply:
Is this an example of what you mean?
```
len(x for x in [(1,2), (1,4), (5,6)])
```

If so I get this:
```
TypeError: object of type 'generator' has no len()
```

Reply:
my bad